centrifuge/client.py

```diff
@@ -451,23 +514,31 @@ async def _sub_refresh(self, channel: str):
         try:
             token = await sub._get_token(SubscriptionTokenContext(channel=channel))
         except Exception as e:
-            if isinstance(e, Unauthorized):
+            if isinstance(e, UnauthorizedError):
                 code = _UnsubscribedCode.UNAUTHORIZED
-                await sub._move_unsubscribed(_code_number(code), _code_message(code), send_unsubscribe_command=True)
+                await sub.move_unsubscribed(
```

Comment:
It was intentionally called with `_` in the beginning to not make part of public API. So it's better to revert this and similar changes (`_resubscribe`, etc). Overall Client and Subscription are tightly coupled – which is not good, but decoupling them is quite hard.